operator/pkg/gateway-api/grpcroute.go

```diff
@@ -62,7 +62,8 @@ func (r *grpcRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	builder := ctrl.NewControllerManagedBy(mgr).
 		// Watch for changes to GRPCRoute
-		For(&gatewayv1.GRPCRoute{}).
+		For(&gatewayv1.GRPCRoute{},
+			builder.WithPredicates(predicate.NewPredicateFuncs(r.hasMatchingGatewayParent()))).
```

Comment:
With the Gateway Parent logic moving into the builder predicates, I'm no longer sure where to put tests for this condition. Would be open to ideas here - maybe writing a quick+dirty unit test for the `hasMatchingGatewayParent` function would be easiest?

Reply:
sounds good to me :heavy_check_mark:

Reply:
Just pushed `operator/pkg/gateway-api/httproute_test.go` with those tests.